esp/platform/esptrace.h

```diff
@@ -0,0 +1,73 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2024 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#pragma once
+
+#include "jtrace.hpp"
+#include <initializer_list>
+
+// Refer to helm/examples/esp/README.md for more information on the trace options.
+
+constexpr const char* propTraceFlags = "traceFlags";
+constexpr const char* propTraceLevel = "traceLevel";
+
+// ESP process names for use with options mapping
+constexpr const char* eclwatchApplication = "eclwatch";
+constexpr const char* eclservicesApplication = "eclservices";
+constexpr const char* eclqueriesApplication = "eclqueires";
+constexpr const char* esdlsandboxApplication = "esdl-sandbox";
+constexpr const char* esdlApplication = "esdl";
+constexpr const char* sql2eclApplication = "sql2ecl";
+constexpr const char* dfsApplication = "dfs";
+constexpr const char* ldapenvironmentApplication = "ldapenvironment";
+constexpr const char* loggingserviceApplication = "loggingservice";
+
+// Trace options for ESP
+constexpr TraceFlags traceLevel = TraceFlags::LevelMask;
```

Comment:
This looks is confusing.  Why have a constant called traceLevel which is actually a bit mask?  It is likely to lead to confusion if there was code which said
if (traceLevel == ....)